### PR TITLE
speed: Always reset the outlen when calling EVP_PKEY_derive

### LINKED ISSUE #18768: openssl 3.0.3: error when running openssl speed ffdh
When running openssl speed with the  command, openssl speed -seconds 1 -multi 256 ffdh, the following errors are reported:
40E747F2327F0000:error:1C80006A:Provider routines:dh_plain_derive:output buffer too small:../providers/implementations/exchange/dh_exch.c:162:

If the value for multi is reduced to around 30, the errors are not reported.

attached:log for command
[command.log](https://github.com/openssl/openssl/files/9077685/command.log)

### PR DESCRIPTION
**Pull request by t8m**
Fixes #18768

### CODE REVIEW

File: apps/speed.c [hunk 1]
```diff
@@ -879,11 +879,14 @@ static int FFDH_derive_key_loop(void *args)
     loopargs_t *tempargs = *(loopargs_t **) args;
     EVP_PKEY_CTX *ffdh_ctx = tempargs->ffdh_ctx[testnum];
     unsigned char *derived_secret = tempargs->secret_ff_a;
-    size_t outlen = MAX_FFDH_SIZE;
     int count;
 
-    for (count = 0; COND(ffdh_c[testnum][0]); count++)
+    for (count = 0; COND(ffdh_c[testnum][0]); count++) {
+        /* outlen can be overwritten with too small value (no padding used) */
```
Comment (paulidale) [inline apps/speed.c:885]: _too small **a** value_.

**Review by paulidale — APPROVED** [review]
Just a nit in a comment :)

**Review by slontis — APPROVED** [review]

**Comment by slontis** [comment]
Approved still if the comment is updated

**Comment by hlandau** [comment]
Merged to master and 3.0. Thank you.

**Pull request closed without merging**